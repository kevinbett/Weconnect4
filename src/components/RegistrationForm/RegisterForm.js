import React from 'react';
import './style.css';


class RegisterForm extends React.Component {
  state = {
    message: undefined
  }

  // Fucntion to handle registration
  handleUserRegister = async (event) => {
    event.preventDefault();
    const name = event.target.username.value;
    const email = event.target.emailid.value;
    const password = event.target.password.value;
    const apiData = await fetch('https://weconnect4-heroku.herokuapp.com/register', {
          method: "POST",
          headers: {'Accept': 'application/json','Content-Type': 'application/json'},
          body: JSON.stringify({
            "name": name,
            "email": email,
            "password": password

          })
    });

      const res = await apiData.json();
      console.log(res)
      
      if (res.message === "User has been registered successfully") {
        window.localStorage.setItem("sMessage", res.message)
        // Redirects to login page
        window.location.assign('/login')
      };
      this.setState({message: res.message, error: res.error});
    }

  render() {
    return (
    <div id="main-registration-container">
     <div id="register">
        <h3>User Registration</h3>
        <form onSubmit= {this.handleUserRegister} >
        <p className="errorMsg">{this.state.message}</p>
        <label>Name</label>
        <input type="text" name="username"  />
        <label>Email ID:</label>
        <input type="text" name="emailid" />
        <label>Password</label>
        <input type="password" name="password" />

        <input type="submit" className="button"  value="Register"/>
        </form>
    </div>
</div>

      );
  }


}


export default RegisterForm;