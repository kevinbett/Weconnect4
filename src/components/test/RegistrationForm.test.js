import RegisterForm from "../RegistrationForm/RegisterForm"
import { shallow } from "enzyme"

describe('Register page renders',() => { 
    it ('populates register form', () => { 
        expect(shallow(<RegisterForm />)).toMatchSnapshot()
    });
} )