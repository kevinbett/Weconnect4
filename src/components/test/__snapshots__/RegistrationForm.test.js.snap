// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register page renders populates register form 1`] = `
<div
  id="main-registration-container"
>
  <div
    id="register"
  >
    <h3>
      User Registration
    </h3>
    <form
      onSubmit={[Function]}
    >
      <p
        className="errorMsg"
      />
      <label>
        Name
      </label>
      <input
        name="username"
        type="text"
      />
      <label>
        Email ID:
      </label>
      <input
        name="emailid"
        type="text"
      />
      <label>
        Password
      </label>
      <input
        name="password"
        type="password"
      />
      <input
        className="button"
        type="submit"
        value="Register"
      />
    </form>
  </div>
</div>
`;
